src/Component/HttpFoundation/src/Websocket/WebsocketResponse.js

```diff
@@ -0,0 +1,249 @@
+const EventSubscriberInterface = Jymfony.Component.EventDispatcher.EventSubscriberInterface;
+const Response = Jymfony.Component.HttpFoundation.Response;
+const CloseFrameException = Jymfony.Component.HttpFoundation.Websocket.Exception.CloseFrameException;
+const Frame = Jymfony.Component.HttpFoundation.Websocket.Frame;
+const FrameBuffer = Jymfony.Component.HttpFoundation.Websocket.FrameBuffer;
+const Message = Jymfony.Component.HttpFoundation.Websocket.Message;
+const HttpServerEvents = Jymfony.Component.HttpServer.Event.HttpServerEvents;
+const crypto = require('crypto');
+
+/**
+ * @memberOf Jymfony.Component.HttpFoundation.Websocket
+ */
+class WebsocketResponse extends mix(Response, EventSubscriberInterface) {
+    /**
+     * Constructor.
+     *
+     * @param {Jymfony.Component.HttpFoundation.Websocket.Websocket} websocket
+     * @param {Object.<string, string|string[]>} headers
+     */
+    __construct(websocket, headers = {}) {
+        super.__construct(undefined, Response.HTTP_SWITCHING_PROTOCOLS, headers);
+
+        /**
+         * @type {net.Socket}
+         *
+         * @private
+         */
+        this._socket = undefined;
+
+        /**
+         * @type {boolean}
+         *
+         * @private
+         */
+        this._connected = false;
+
+        /**
+         * @type {Jymfony.Component.HttpFoundation.Websocket.FrameBuffer}
+         *
+         * @private
+         */
+        this._frameBuffer = new FrameBuffer();
+
+        /**
+         * @type {Jymfony.Component.HttpFoundation.Websocket.Websocket}
+         *
+         * @private
+         */
+        this._websocket = websocket;
+
+        /**
+         * @type {Jymfony.Component.EventDispatcher.EventDispatcherInterface}
+         *
+         * @private
+         */
+        this._eventDispatcher = undefined;
+    }
+
+    /**
+     * Subscribe to the given event dispatcher.
+     *
+     * @param {Jymfony.Component.EventDispatcher.EventDispatcherInterface} dispatcher
+     */
+    set eventDispatcher(dispatcher) {
+        this._eventDispatcher = dispatcher;
+        dispatcher.addSubscriber(this);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    prepare(request) {
+        super.prepare(request);
+
+        this.headers.set('Upgrade', 'websocket');
+        this.headers.set('Connection', 'upgrade');
+
+        const sha1 = crypto.createHash('sha1');
+        sha1.update(request.headers.get('Sec-WebSocket-Key') + '258EAFA5-E914-47DA-95CA-C5AB0DC85B11');
```

Comment:
Can maybe this guid set into a constant of `WebsocketResponse`? The constant can be named with something like `RFC6455_GUID`